ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Types.hs

```diff
@@ -414,7 +423,14 @@ data PeerSelectionState peeraddr peerconn = PeerSelectionState {
        -- | 'PeerSelectionCounters' counters cache. Allows to only trace
        -- values when necessary.
        --
-       countersCache               :: !(Cache PeerSelectionCounters)
+       countersCache               :: !(Cache PeerSelectionCounters),
+
+       -- | Cached value of 'LedgerStateJudgement'. If this changes the peer selection
+       -- governor will act on it
+       currentLedgerStateJudgement :: !LedgerStateJudgement,
```

Comment:
We need to cache it because that's the best way to guard the monitoring jobs against this value, otherwise we'd need to waste governor's time checking it at "runtime".